rocketmq-spring-boot/src/main/java/org/apache/rocketmq/spring/config/ListenerContainerConfiguration.java

```diff
@@ -42,6 +42,7 @@
 import java.util.concurrent.atomic.AtomicLong;
 
 
+@Configuration
```

Comment:
Is this annotation optional ? since it is imported (@Import) by root configuration class: RocketMQAutoConfiguration.java

@snicoll

Reply:
It's not optional. Adding `@Configuration` flags this class as a configuration class. `@Import` is generic, you could `@Import` a regular class and it will be exposed as a bean.

Adding `@Configuration` is important as the `ApplicationContext` handles classes with the `@Configuration` stereotype differently than regular classes. 

The most typical example is proxying mode where you pass references around using method invocation, e.g.


```
@Configuration
public Class MyConfig {

    @Bean
    public MyBean first() { ... }

    public AnotherBean second() {
       return new AnotherBean(first());
    }
}
```

If you remove `@Configuration` the internal `first()` call in `AnotherBean` is not going to return the same bean instance as the bean exposed by the `@Bean` method.

Removing `@Configuration` turns your class in lite mode (which I'd like to get rid of at some point to be honest).